### DISCUSSION TITLE
Occasionally open the previous file by mistake.

### DISCUSSION THREAD

**Discussion by tyf2018** [Taitava/obsidian-shellcommands#413] [Debugging and testing]
### Obsidian debug information

SYSTEM INFO:
	Obsidian version: v1.6.1
	Installer version: v1.5.8
	Operating system: Windows 10 Pro 10.0.19045
	Login status: logged in
	Catalyst license: insider
	Insider build toggle: on
	Live preview: on
	Base theme: dark
	Community theme: Red_Graphite改良 v20230223
	Snippets enabled: 6
	Restricted mode: off
	Plugins installed: 31
	Plugins enabled: 21
		1: Reminder v1.1.15
		2: Commander v0.5.1
		3: Hotkeys++ v0.2.7
		4: Templater v2.3.2
		5: Projects v1.17.3
		6: QuickAdd v1.9.0
		7: Sortable v0.3.1
		8: Another Quick Switcher v11.4.0
		9: Various Complements v10.0.0
		10: Shell commands v0.22.0
		11: AttachFlow v0.4.3
		12: Better Word Count v0.10.1
		13: Hotkeys for specific files v1.4.1
		14: Dataview v0.5.66
		15: Publish url v1.0.8
		16: Text Generator v0.7.11
		17: Quick Explorer v0.2.8
		18: Footnote Shortcut v0.1.3
		19: 增强编辑 v0.6.0
		20: Linter v1.24.0
		21: Easy Typing v5.3.3

RECOMMENDATIONS:
	Custom theme and snippets: for cosmetic issues, please first try updating your theme and disabling your snippets. If still not fixed, please try to make the issue happen in the Sandbox Vault or disable community theme and snippets.
	Community plugins: for bugs, please first try updating all your plugins to latest. If still not fixed, please try to make the issue happen in the Sandbox Vault or disable community plugins.

### What shell are you using?

PowerShell Core

### Please describe the problem

1. Open File A
2. Switch the current tab to File B
3. Run the command：`C:\totalcmd\Tools\Emeditor\EmEditor.exe "{{file_path:absolute}}" /l {{caret_position}}`
4. The expected result should be opening File B in EmEditor. The actual result: most of the time it works correctly, but occasionally it opens File A instead of File B.

Possible Reason: My vault  is quite large, with over ten thousand notes, which may be causing delays in data retrieval.

**Comment by Taitava** [comment]
Hi @tyf2018 and thank you for your message! 🙂 

This problem looks similar to #379. The discussion is ongoing and the reason for this problem is still unknown. As your use-case is a bit different than in #379 (you don't seem to be using QuickAdd to execute the shell command), we can continue this discussion with you here in this #413.

I'd like to ask you similar things as I asked from @ces3001 in #379:
- Can you please record a video of the problem occurring? I understand that this can be a bit hard as the problem does not occur every time.
- Can you please also share the _Shell commands_ plugin's settings file (`.obsidian/plugins/obsidian-shellcommands/data.json`)?

If you don't like sharing these things, it's okay, we can try to find other ways to research this problem. 👍

**Comment by tyf2018** [comment]
Thanks for your answer.It is indeed difficult to record a video. However, this is a configuration file. I post it first.

[data.json](https://github.com/user-attachments/files/15571656/data.json)
